validator/lib/validateBatchTable.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+var Ajv = require('ajv');
+var Cesium = require('cesium');
+
+var defined = Cesium.defined;
+
+module.exports = validateBatchTable;
+
+var componentTypeSize = {
+    'BYTE': 1,
+    'UNSIGNED_BYTE': 1,
+    'SHORT': 2,
+    'UNSIGNED_SHORT': 2,
+    'INT': 4,
+    'UNSIGNED_INT': 4,
+    'FLOAT': 4,
+    'DOUBLE': 8
+};
+
+var typeSize = {
+    'SCALAR': 1,
+    'VEC2': 2,
+    'VEC3': 3,
+    'VEC4': 4
+};
+
+/**
+ * Checks if provided buffers follow the batch table schema
+ *
+ * @param {Object} schema - A JSON object containing the schema for the batch table.
+ * @param {JSON} batchTableJSON - Batch table JSON
+ * @param {Buffer} batchTableBinary - A buffer containing the batch table binary
+ * @returns {Object} An object with two parameters - (1) a boolean for whether the batch table follows the schema
+ *                                                   (2) a message to indicate the validation result
+ */
+function validateBatchTable(schema, batchTableJSON, batchTableBinary) {
+    var ajv = new Ajv();
+    var validSchema = ajv.validate(schema, batchTableJSON);
+    var valid = validSchema;
+    var message;
+
+    if(validSchema) {
+        if(defined(batchTableBinary) && (batchTableBinary.length > 0)) {
+            var binaryBodyLength = batchTableBinary.length;
+            var batchLength = batchTableJSON["id"].length;
```

Comment:
The property `id` is not guaranteed to exist. The batch length is now in the feature table: https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/master/TileFormats/Batched3DModel#global-semantics